content/sodiq_aderibigbe/creating-queues-using-javascript

```diff
@@ -0,0 +1,154 @@
+---
+Title: "Creating Queues Using JavaScript"
+Description: "Creating a queue and implementing several of it's operations using JavaScript"
+DatePublished: "2022-04-28"
+Categories:
+  - "computer-science"
+Tags:
+  - "Algorithms"
+  - "Arrays"
+  - "ES6"
+  - "Classes"
+CatalogContent:
+  - "learn-javascript"
+  - "course/linear-data-structures"
+---
+
+## Creating Queues Using JavaScript
+
+During conversations about data structures in the programming world, queues are heard quite often.
+A queue is a linear data structure that follows the first-in-first-out pattern i.e removal takes place at the front, and addition takes place at the end. Think of it as the checkout point of a grocery store. Customers who walk there first get attended to first.
+
+
+## Use cases for queues
+
+There are several use cases for queues in the programming and real world. some them include:
+
+- A desk printer 
+- Adding songs to a queue in your music player
+- Customer service wait-lines
+- File sharing between / data transfer between two processes
```

Comment:
Maybe we tack on a colon ":" to each of these list items and go into a little bit of detail about how these are related to queues!